back/boxtribute_server/models/crud.py

```diff
@@ -65,6 +76,161 @@ def create_box(
     raise BoxCreationFailed()
 
 
+def move_items_from_box_to_distribution_event(
+    user_id, box_label_identifier, distribution_event_id, number_of_items
+):
+    """
+    Move items from a box to a distribution event.
+    """
+
+    with db.database.atomic():
+        # Completed Events should not be mutable anymore
+        distribution_event = DistributionEvent.get_by_id(distribution_event_id)
+        if distribution_event.state == DistributionEventState.Completed:
+            raise GraphQLError("Cannot move items to completed distribution event")
+
+        box = Box.get(Box.label_identifier == box_label_identifier)
+
+        # TODO: Discuss error handling approach:
+        # Ok to throw GraphQL errors in the crud module?
```

Comment:
You could define a custom exception from the `exceptions` module (maybe re-use existing base exceptions). Then raise it here. ariadne will wrap it into a GraphQL error

Reply:
👍